src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/MagicNumberCheck.java

```diff
@@ -515,9 +515,10 @@ private static boolean isFieldDeclaration(DetailAST ast) {
         }
 
         // contains variable declaration
-        // and it is directly inside class declaration
+        // and it is directly inside class or record declaration
         return varDefAST != null
-                && varDefAST.getParent().getParent().getType() == TokenTypes.CLASS_DEF;
+                && (varDefAST.getParent().getParent().getType() == TokenTypes.CLASS_DEF
+                || varDefAST.getParent().getParent().getType() == TokenTypes.RECORD_DEF);
```

Comment:
please add code example for record in doc

Reply:
Done. Since there wasn't an example for ignoreFieldDeclaration , I added an example of that config usage in a record definition.